WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.kt

```diff
@@ -0,0 +1,130 @@
+package org.wordpress.android.ui.posts
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.appcompat.view.ContextThemeWrapper
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.databinding.AddCategoryBinding
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.TermModel
+import org.wordpress.android.fluxc.store.TaxonomyStore
+import org.wordpress.android.models.CategoryNode
+import org.wordpress.android.util.ToastUtils
+import javax.inject.Inject
+
+class AddCategoryFragment : AppCompatDialogFragment() {
+    private var site: SiteModel? = null
+    private var binding: AddCategoryBinding? = null
+
+    @set:Inject
+    var mTaxonomyStore: TaxonomyStore? = null
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        (requireActivity().application as WordPress).component().inject(this)
+        initSite(savedInstanceState)
+        val builder =
+            MaterialAlertDialogBuilder(ContextThemeWrapper(activity, R.style.PostSettingsTheme))
+        binding = AddCategoryBinding.inflate(layoutInflater, null, false)
+        loadCategories()
+        builder.setView(binding!!.root)
```

Comment:
I'm thinking of a way of getting rid of !!. If we view null with ` builder.setView(binding?.root)`, we can avoid !! crash. I haven't tested it but it will probably show a blank page instead of !! crash.